ouroboros-network-framework/src/Ouroboros/Network/InboundGovernor/ControlChannel.hs

```diff
@@ -57,50 +56,43 @@ instance Show peerAddr
 
 -- | Server control channel.  It allows to pass 'STM' transactions which will
 -- resolve to 'NewConnection'.   Server's monitoring thread is the consumer
--- of this messages; there are two produceres: accept loop and connection
+-- of these messages; there are two producers: accept loop and connection
 -- handler for outbound connections.
 --
+-- GR-FIXME[D2]: Last sentence is DOU (see code-review-doc)
+type ServerControlChannel (muxMode :: MuxMode) peerAddr bytes m a b =
+    ControlChannel m (NewConnection peerAddr (Handle muxMode peerAddr bytes m a b))
+
 data ControlChannel m msg =
   ControlChannel {
-    -- | Read a single 'NewConnection' instructrion from the channel.
+    -- | Read a single 'NewConnection' instruction from the channel.
     --
     readMessage  :: STM m msg,
 
     -- | Write a 'NewConnection' to the channel.
     --
     writeMessage :: msg -> STM m ()
   }
-
-
-type ServerControlChannel (muxMode :: MuxMode) peerAddr bytes m a b =
-    ControlChannel m (NewConnection peerAddr (Handle muxMode peerAddr bytes m a b))
+  -- GR-FIXME[C2]: the 'msg' tyvar appears to be always instantiated to
+  --   'NewConnection ...'
+  --   i.e., unnecessary polymorphism: reason for? to avoid dup of (NewConnection ...)?
 
 
 newControlChannel :: forall m srvCntrlMsg.
                      MonadLabelledSTM m
                   => m (ControlChannel m srvCntrlMsg)
 newControlChannel = do
-    -- Queue size: events will come eihter from the accept loop or from the
-    -- connection manager (when it included an outbound duplex connection).
+    -- Queue size: events will come either from the accept loop or from the
+    -- connection manager (when it includes an outbound duplex connection).
+    -- GR-FIXME[D2]: above comment is DOU (see code-review-doc)
```

Comment:
This comment actually duplicates the information given in `ServerControlChannel` and thus can be removed.